src/core/backbone_router/local.cpp

```diff
@@ -238,6 +251,90 @@ void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRout
     return;
 }
 
+otError Local::GetDomainPrefix(NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_NOT_FOUND);
+
+    aConfig = mDomainPrefixConfig;
+
+exit:
+    return error;
+}
+
+otError Local::RemoveDomainPrefix(const otIp6Prefix &aPrefix)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength == aPrefix.mLength, error = OT_ERROR_NOT_FOUND);
+
+    VerifyOrExit(otIp6PrefixMatch(&mDomainPrefixConfig.mPrefix.mPrefix, &aPrefix.mPrefix) >=
+                     mDomainPrefixConfig.mPrefix.mLength,
+                 error = OT_ERROR_NOT_FOUND);
+
+    mDomainPrefixConfig.mPrefix.mLength = 0;
+
+    if (IsEnabled())
+    {
+        SuccessOrExit(error =
+                          Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength));
+    }
+
+exit:
+    return error;
+}
+
+otError Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aConfig.mPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    if (IsEnabled())
+    {
+        Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+                                                     mDomainPrefixConfig.mPrefix.mLength);
+    }
+
+    mDomainPrefixConfig = aConfig;
+
+    if (IsEnabled())
+    {
+        SuccessOrExit(error = Get<NetworkData::Local>().AddOnMeshPrefix(mDomainPrefixConfig));
+    }
+
+exit:
+    return error;
+}
+
+otError Local::RemoveDomainPrefix(void)
```

Comment:
I think it would be more clear if we do not overload the method name. Can we have something like:
- `RemoveDomainPrefixFromNetworkData()`
- `RemoveDomainPrefix()`

Something similar with `AddPrefix()`?